voip_utils/sip.py

```diff
@@ -86,12 +152,12 @@ def datagram_received(self, data: bytes, addr):
 
             caller_uri = None
             caller_name = None
-            # The Contact header should give us the URI used for sending SIP messages to the device
-            if headers.get("contact") is not None:
-                caller_uri, caller_name = self._parse_uri_header(headers.get("contact"))
-            # We can try using the From header as a fallback
-            elif headers.get("from") is not None:
+            # The From header should give us the URI used for sending SIP messages to the device
+            if headers.get("from") is not None:
```

Comment:
From the SIP spec it looked liked the Contact header was what was supposed to be used as the URI for an individual endpoint but from testing with the Grandstream and calls through Asterisk it appears the From header actually contains the URI needed to contact the endpoint.

Reply:
Opened a separate PR just for this https://github.com/home-assistant-libs/voip-utils/pull/20